### PR TITLE
PreImageCycle: Allow creation from a PI + height

### PR DESCRIPTION
**Pull request by Geod24**
This will allow us to extend the default number of pre-images to a very large number,
while still keeping a limited number of pre-images in the test network.

### CODE REVIEW

File: source/agora/consensus/PreImage.d [hunk 1]
```diff
@@ -291,17 +291,71 @@ public struct PreImageCycle
...
+
     public this (in Scalar secret, in uint cycle_length, in ulong cycles = PreImageCycle.NumberOfCycles,
-        Height initial_seek = Height(0))
+        in Height initial_seek = Height(0))
+    {
+        // Using this predictable scheme ensures that the pre-image can
+        // be recovered in case of a crash or if another node takes over
+        // (e.g. disk failure from a server leads to a backup server starting
+        // with no state or way to recover it).
+        const cycle_seed = hashMulti(secret, "consensus.preimages", 0);
```
Comment (hewison-chris) [inline source/agora/consensus/PreImage.d:320]: Do we need to keep this `nonce` of zero still?

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/agora/consensus/PreImage.d (hunk 1 above)
Comment (Geod24) [reply source/agora/consensus/PreImage.d:320]: Not that I can think of. The only upside is to be consistent with how other parameters are computed.

**Review by linked0 — APPROVED** [review]
LGTM. I can use the second constructor in my PR. Ty.

File: source/agora/consensus/PreImage.d (hunk 1 above)
Comment (linked0) [reply source/agora/consensus/PreImage.d:320]: It's not necessary. I will remove it which should have been removed in the recent commit I have submitted.

**Review by kchulj — APPROVED** [review]

**Pull request merged**